.github/workflows/ci.yml

```diff
@@ -0,0 +1,96 @@
+# See https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions
+
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  Tests:
+    runs-on: ubuntu-latest
+    container: registry.opensuse.org/yast/head/containers/yast-ruby:latest
+
+    steps:
+
+    - name: Git Checkout
+      uses: actions/checkout@v2
+
+    # just for easier debugging...
+    - name: Inspect Installed Packages
+      run: rpm -qa | sort
+
+    - name: Unit Tests
+      run: rake test:unit
+      # enable code coverage reporting
+      env:
+        COVERAGE: 1
+
+    # send the coverage report to coveralls.io
+    - name: Coveralls Report
+      uses: coverallsapp/github-action@master
+      with:
+        github-token: ${{ secrets.GITHUB_TOKEN }}
+
+  Rubocop:
+    runs-on: ubuntu-latest
+    container: registry.opensuse.org/yast/head/containers/yast-cpp:latest
+
+    steps:
+
+    - name: Git Checkout
+      uses: actions/checkout@v2
+
+    - name: Rubocop
+      run: rake check:rubocop
+
+  Package:
+    runs-on: ubuntu-latest
+    container: registry.opensuse.org/yast/head/containers/yast-cpp:latest
+
+    steps:
+
+    - name: Git Checkout
+      uses: actions/checkout@v2
+
+    - name: Prepare System
+      run: |
+        # build_dependencies:install is not enough as some perl deps are runtime only
+        rake build_dependencies:install
+        zypper --non-interactive in --force-resolution --no-recommends \
+          cracklib-devel \
+          perl-Digest-SHA1 \
+          perl-X500-DN \
+          yast2-perl-bindings
+
+    - name: Package Build
+      run: yast-ci-cpp
+      env:
+        # the tests require specific locale settings to pass
+        LANG: POSIX
+        LC_ALL: ""
+        # for the Perl syntax check
+        PERL5LIB: "./agents-perl/lib"
```

Comment:
This should be probably moved to the Perl syntax check step, see below.

Reply:
it is done as part of yast-ci-cpp. If we do it below, we have it also below force installation of that perl packages, otherwise perl check failed. So it I think it is better to have it in this one task which also include packages needed for build.

Reply:
Ah, I see, then it makes sense to run it here...